src/Avalonia.ReactiveUI/RoutedViewHost.cs

```diff
@@ -67,12 +67,12 @@ public RoutedViewHost()
             {
                 var routerRemoved = this
                     .WhenAnyValue(x => x.Router)
-                    .Where(router => router == null)
-                    .Cast<object>();
+                    .Where(router => router == null)!
```

Comment:
Not sure why it doesn't allow to cast from Router? to object?.

Reply:
Looks like yet another nullable reference inference bug that will be hopefully fixed in future .NET releases.

Worth noting that the `IObservable<T>` can never be null after a `Where` call.